Samples/Client/Unity/Assets/Scripts/ControlScript.cs

```diff
@@ -213,22 +210,22 @@ void Update()
             for (int j = 0; j < 4; j++)
             {
                 leftCameraTransform += leftViewProjection[i, j] + ",";
-                rightCameraTransform += rightViewProjection[i, j];
-                if (i != 3 || j != 3)
-                {
-                    rightCameraTransform += ",";
-                }
+                rightCameraTransform += rightViewProjection[i, j] + ",";
             }
         }
 
         var cameraTransformBody = leftCameraTransform + rightCameraTransform;
-        cameraTransformMsg =
+
+        // Adds dummy prediction timestamp.
+        cameraTransformBody += timestamp++;

```

Comment:
this will break backward compatibility with old server implementations, correct? can we instead use a different message schema for this so that new clients (running this code) can talk to both 1) old clients (that don't understand timestamps in the body) and 2) new clients (that do understand timestamps)

I'd propose a `timestamp` field in addition to `type` and `body`.

Reply:
No, it doesn't since it's appended as the last element. Older versions won't just parse it. The timestamp here is actual the prediction timestamp, which is used for frame prediction, so it should be part of the body message. Additionally, we should encourage developers to use the latest Unity server/client since it has several performance optimizations (even for mono mode).